Src/ILGPU/IR/BasicBlock.cs

```diff
@@ -453,8 +453,12 @@ public bool HasSideEffects()
         {
             foreach (Value value in this)
             {
-                if (value is MemoryValue || value is MethodCall)
+                if (value is MemoryValue ||
+                    value is MethodCall ||
+                    value is LanguageEmitValue)
```

Comment:
If we change the base type to `MemoryValue` we can safely remove this again 🤓

Reply:
Done.